java/cdk-custom-resources/src/test/java/sleeper/cdk/custom/WiremockEmrPersistentTestHelper.java

```diff
@@ -0,0 +1,150 @@
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.matching;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static sleeper.localstack.test.WiremockAwsV2ClientHelper.wiremockAwsV2Client;
+
+/**
+ * Helper methods to mock an EMR Persistent cluster.
+ */
+public class WiremockEmrPersistentTestHelper {
+    private WiremockEmrPersistentTestHelper() {
+    }
+
+    public static final String OPERATION_HEADER = "X-Amz-Target";
+    private static final StringValuePattern MATCHING_DESCRIBE_CLUSTER_OPERATION = equalTo("ElasticMapReduce.DescribeCluster");
+    private static final StringValuePattern MATCHING_TERMINATE_JOB_FLOWS_OPERATION = equalTo("ElasticMapReduce.TerminateJobFlows");
+
+    /**
+     * Creates a mocked EMR Servless client.
+     *
+     * @param  runtimeInfo wire mocks runtime info
+     * @return             the EMR Serverless client
+     */
+    public static EmrClient wiremockEmrClient(WireMockRuntimeInfo runtimeInfo) {
+        return wiremockAwsV2Client(runtimeInfo, EmrClient.builder());
+    }
+
+    /**
+     * Check for any requests to the EMR client.
+     *
+     * @return matching HTTP requests
+     */
+    public static RequestPatternBuilder anyRequestedForEmr() {
+        return anyRequestedFor(anyUrl())
+                .withHeader(OPERATION_HEADER, matching("^ElasticMapReduce\\..*"));
+    }
+
+    /**
+     * Build an EMR clustern response with a given cluster id and state.
+     *
+     * @param  clusterId the cluster id
+     * @param  state     the cluster state
+     * @return           a HTTP response
+     */
+    public static ResponseDefinitionBuilder aResponseWithClusterWithState(String clusterId, ClusterState state) {
+        return aResponse().withStatus(200).withBody("{\"Cluster\": {" +
+                "\"Id\":\"" + clusterId + "\"," +
+                "\"Name\":\"sleeper-test-instance-test-cluster\"," +
+                "\"Status\": {\"State\": \"" + state.toString() + "\"}" +
+                "}}");
+    }
+
+    /**
+     * Describe an EMR cluster.
+     *
+     * @param  clusterId the cluster id
+     * @return           a HTTP response
+     */
+    public static MappingBuilder describeClusterRequest(String clusterId) {
+        return post("/")
+                .withHeader(OPERATION_HEADER, MATCHING_DESCRIBE_CLUSTER_OPERATION)
+                .willReturn(aResponse().withStatus(200));
```

Comment:
This isn't using the cluster ID parameter passed in. We probably don't need to pass it in, as we're asserting the body of this request with describeClusterRequestedFor(clusterId).